contracts/smartdeploy/src/registry/contract.rs

```diff
@@ -96,14 +89,14 @@ impl IsDeployable for ContractRegistry {
             },
             Ok,
         )?;
-        let deploy_datas = DeployEventData {
+        let deploy_data = Deploy {
             published_name: contract_name,
             deployed_name,
             version,
             deployer: owner,
             contract_id: address.clone(),
         };
```

Comment:
You can call the method without needing to bind to local variable.
```suggestion
        }.publish_event(env())
```